src/mixins/changeStateMixin.js

```diff
@@ -131,13 +133,35 @@ export const changeStateMixin = {
         'An error occurred when approving this task run. Please try again.'
       this.changeState()
     },
-    cancelFlowRun() {
+    async cancelFlowRun() {
       this.cancelLoad = true
       this.selectedState = 'Cancelled'
       this.allTasks = true
-      this.toastedMessage =
-        'Please try pressing cancel again. If the flow run has finished, use "Mark As" instead.'
-      this.changeState()
+
+      try {
+        await this.$apollo.mutate({
+          mutation: require('@/graphql/FlowRun/cancel-flow-run.gql'),
+          variables: {
+            flowRunId: this.flowRun.id
+          }
+        })
+
+        this.setAlert({
+          alertShow: true,
+          alertMessage:
+            'Your flow run has been cancelled; please allow up to 30 seconds for this to take effect.',
```

Comment:
We might want to change the wording here -> it will take up to 30 seconds for their workflow to learn of the desire for cancellation, and then another indeterminate amount of time to actually cancel (depending on the type of work being run, etc.).  Maybe 
```
Your flow run will be cancelled; please allow some time for this to take effect.
```